local/src/main/scala/com/salesforce/op/local/MLeapModelConverter.scala

```diff
@@ -0,0 +1,92 @@
+   */
+  def modelToFunction(model: Any): Array[Any] => Any = model match {
+    case m: BinarizerModel => x => m.apply(x(0).asInstanceOf[Number].doubleValue())
+    case m: BucketedRandomProjectionLSHModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: BucketizerModel => x => m.apply(x(0).asInstanceOf[Number].doubleValue())
+    case m: ChiSqSelectorModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: CoalesceModel => x => m.apply(x: _*)
+    case m: CountVectorizerModel => x => m.apply(x(0).asInstanceOf[Seq[String]])
+    case m: DCTModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: ElementwiseProductModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: FeatureHasherModel => x => m.apply(x(0).asInstanceOf[Seq[Any]])
+    case m: HashingTermFrequencyModel => x => m.apply(x(0).asInstanceOf[Seq[Any]])
+    case m: IDFModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: ImputerModel => x => m.apply(x(0).asInstanceOf[Number].doubleValue())
+    case m: InteractionModel => x => m.apply(x(0).asInstanceOf[Seq[Any]])
+    case m: MathBinaryModel => x =>
+      m.apply(
+        x.headOption.map(_.asInstanceOf[Number].doubleValue()),
+        x.lastOption.map(_.asInstanceOf[Number].doubleValue())
+      )
+    case m: MathUnaryModel => x => m.apply(x(0).asInstanceOf[Number].doubleValue())
+    case m: MaxAbsScalerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: MinHashLSHModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: MinMaxScalerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: NGramModel => x => m.apply(x(0).asInstanceOf[Seq[String]])
+    case m: NormalizerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: OneHotEncoderModel => x => m.apply(x(0).asInstanceOf[Vector].toArray)
+    case m: PcaModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: PolynomialExpansionModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: RegexIndexerModel => x => m.apply(x(0).toString)
+    case m: RegexTokenizerModel => x => m.apply(x(0).toString)
+    case m: ReverseStringIndexerModel => x => m.apply(x(0).asInstanceOf[Number].intValue())
+    case m: StandardScalerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: StopWordsRemoverModel => x => m.apply(x(0).asInstanceOf[Seq[String]])
+    case m: StringIndexerModel => x => m.apply(x(0))
+    case m: StringMapModel => x => m.apply(x(0).toString)
+    case m: TokenizerModel => x => m.apply(x(0).toString)
+    case m: VectorAssemblerModel => x => m.apply(x(0).asInstanceOf[Seq[Any]])
+    case m: VectorIndexerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: VectorSlicerModel => x => m.apply(x(0).asInstanceOf[Vector])
+    case m: WordLengthFilterModel => x => m.apply(x(0).asInstanceOf[Seq[String]])
+    case m: WordToVectorModel => x => m.apply(x(0).asInstanceOf[Seq[String]])
+    case m => throw new RuntimeException(s"Unsupported MLeap model: ${m.getClass.getName}")
```

Comment:
so every wrapped spark stage has to be in this list? we should add that the the docs on wrapping...

Reply:
I currently added all the stages from [features package](https://github.com/combust/mleap/tree/master/mleap-core/src/main/scala/ml/combust/mleap/core/feature). We can also add models from [classification](https://github.com/combust/mleap/tree/master/mleap-core/src/main/scala/ml/combust/mleap/core/classification), [regression](https://github.com/combust/mleap/tree/master/mleap-core/src/main/scala/ml/combust/mleap/core/regression) and [recommendation](https://github.com/combust/mleap/tree/master/mleap-core/src/main/scala/ml/combust/mleap/core/recommendation) packages, but we already have the first two of them covered as our own `OpTransformer` stages, so I did not see much of a point adding them.

Reply:
so to your question - for right now I think we have everything covered, except recommenders, which I am planning to add once we are ready.

Reply:
Can you add a todo with the classification and regression models? I dont know that this will be much use without them...

Reply:
Adding those is very easy. the thing is we already have classification and regression models as OpTransformers so MLeap won’t be used to run them.

Reply:
good point :-)